src/jit/codegenarmarch.cpp

```diff
@@ -668,9 +668,10 @@ void CodeGen::genPutArgStk(GenTreePutArgStk* treeNode)
 
                 // Emit store instructions to store the registers produced by the GT_FIELD_LIST into the outgoing
                 // argument area
-                emit->emitIns_S_R(ins_Store(type), attr, reg, varNumOut, argOffsetOut);
-                argOffsetOut += EA_SIZE_IN_BYTES(attr);
-                assert(argOffsetOut <= argOffsetMax); // We can't write beyound the outgoing area area
```

Comment:
Original code was packing the struct on the stack which was incorrect.

Reply:
Looks correct now